### PR TITLE
installer: do not report failure when new revision is waiting

### PR DESCRIPTION
**Pull request by mfojtik**
Instead of reporting synthetic failure when current revision is superseded by new revision, report just event and move to the new revision without any noise.

/cc @deads2k 
/cc @sttts

### CODE REVIEW

**Comment by bparees** [comment]
@mfojtik who do you expect to review/merge this?

**Comment by mfojtik** [comment]
/hold

lets make fake bump and prove this works

File: pkg/operator/staticpod/controller/installer/installer_controller.go [hunk 1]
```diff
@@ -515,19 +515,21 @@ func (c *InstallerController) newNodeStateForInstallInProgress(currNodeState *op
...
+		state, podRevision, failedErrors, err := c.getStaticPodState(currNodeState.NodeName)
 		if err != nil {
 			return nil, false, err
 		}
 
-		if revision != strconv.Itoa(int(currNodeState.TargetRevision)) {
+		// stop early, don't wait for ready static pod because a new revision is waiting
+		if newRevisionPending {
+			ret.TargetRevision = 0
+			ret.LastFailedRevision = 0
```
Comment (sttts) [inline pkg/operator/staticpod/controller/installer/installer_controller.go]: why this?

**Comment by sttts** [comment]
This should lead to test changes. I.e. we need a test for that.

**Pull request closed without merging**